trin-core/src/portalnet/overlay_service.rs

```diff
@@ -917,12 +955,30 @@ impl OverlayService {
 
         Ok(closest_nodes)
     }
+
+    fn send_recursive_findnode(&self, _target: &NodeId) {
+        // TODO: Implement Recursive(iterative) FINDNODE. This is a stub.
+    }
+
+    /// Generate random NodeId based on bucket index target.
+    /// First we generate a random distance metric with leading zeroes based on the target bucket.
+    /// Then we XOR the result distance with the local NodeId to get the random target NodeId
+    fn generate_random_node_id(&self, target_bucket_idx: u8) -> anyhow::Result<NodeId> {
+        let distance_leading_zeroes = 255 - target_bucket_idx;
+        let random_distance = bytes::random_32byte_array(distance_leading_zeroes);
+        let raw_node_id = xor_two_values(&self.local_enr().node_id().raw(), &random_distance)?;
+        match NodeId::parse(raw_node_id.as_slice()) {
+            Ok(node_id) => Ok(node_id),
+            Err(msg) => Err(anyhow::Error::msg(msg)),
+        }
```

Comment:
I think the `anyhow` docs suggest that this has the same effect:
```suggestion
        let parsed_id = NodeId::parse(raw_node_id.as_slice())?;
        Ok(parsed_id)
```

Reply:
Yeah, true but the weird thing is that `NodeId::parse` returns `Result<Self, &'static str> ` instead of `Result<Self, std::error::Error>`, and this force us to handle the string manually...

Reply:
Ah I see, thanks for clarifying. I guess we can add it to the pile of nice-to-have features for discv5.